eyeshade/lib/transaction.js

```diff
@@ -5,13 +5,69 @@ const {
   createdTimestamp,
   normalizeChannel
 } = require('bat-utils/lib/extras-utils')
+const queries = require('./queries')
 
 const SETTLEMENT_NAMESPACE = {
   'contribution': '4208cdfc-26f3-44a2-9f9d-1f6657001706',
   'referral': '7fda9071-4f0d-4fe6-b3ac-b1c484d5601a'
 }
 
-exports.insertFromSettlement = async (runtime, client, settlement) => {
+module.exports = {
+  insertFromSettlement,
+  insertFromVoting,
+  insertFromReferrals,
+  updateBalances,
+  findTransactionByToken,
+  insertFromAd
+}
+
+async function findTransactionByToken (runtime, client, {
+  paymentId,
+  tokenId
+}) {
+  const query = queries.findTransactionByToken()
+  const replacements = [paymentId, tokenId]
+  const { rows } = await client.query(query, replacements)
+  return rows
+}
+
+async function insertFromAd (runtime, client, {
+  paymentId,
+  tokenId,
+  amount
+}) {
+  const query = queries.insertTransaction()
+  const created = seconds()
+  const { config } = runtime
+  const id = uuidv5(`${paymentId}:${tokenId}`, '2ca02950-084f-475f-bac3-42a3c99dec95')
+  const probi = (new BigNumber(amount)).toString()
+  const month = monthsFromSeconds(created)
+  const replacements = [
+    id,
+    created,
```

Comment:
the `created_at` field can simply be omitted and postgres will use the current date as the default per https://github.com/brave-intl/bat-ledger/blob/master/eyeshade/migrations/0001_transactions/up.sql#L10

Reply:
can remove.